src/v/cluster/config_manager.cc

```diff
@@ -0,0 +1,740 @@
+            auto decoded = YAML::Load(value.as<std::string>());
+            cfg.get(key).set_value(decoded);
+
+            // Because we are in preload, it doesn't matter if the property
+            // requires restart.  We are setting it before anything else
+            // can be using it.
+
+            if (result.has_value()) {
+                vlog(
+                  clusterlog.trace,
+                  "Loaded property {}={} from local cache",
+                  key,
+                  YAML::Dump(decoded));
+            }
+        } catch (...) {
+            if (result.has_value()) {
+                vlog(
+                  clusterlog.info,
+                  "Ignoring invalid property: {}={}",
+                  key,
+                  YAML::Dump(value));
+                result.value().get().invalid.push_back(key);
+            }
+        }
+    } else {
+        if (result.has_value()) {
+            vlog(clusterlog.info, "Ignoring unknown property: {}", key);
+            result.value().get().unknown.push_back(key);
+        }
+    }
+}
+
+/**
+ * Before other redpanda subsystems are initialized, read the config
+ * cache and use it to bring the global cluster configuration somewhat
+ * up to date before the raft0 log is replayed.
+ */
+ss::future<config_manager::preload_result> config_manager::preload() {
+    // TODO: refactor yaml load workaround for lack of string view,
+    // it's also used in node_config load.
+
+    // TODO: legacy path: if cache isn't present, load from redpanda-cfg
+    // (currently we rely on application.cc to do this)
+
+    // TODO: assert that node_config was already loaded (we rely on
+    // data_directory)
+
+    preload_result result;
+
+    YAML::Node config;
+    try {
+        auto buf = co_await read_fully(cache_path());
+        auto workaround = ss::uninitialized_string(buf.size_bytes());
+        auto in = iobuf::iterator_consumer(buf.cbegin(), buf.cend());
+        in.consume_to(buf.size_bytes(), workaround.begin());
+        config = YAML::Load(workaround);
+    } catch (std::filesystem::filesystem_error const& e) {
+        // This is normal on upgrade from pre-config_manager version or
+        // on newly added node.  Also permitted later if user
+        // chooses to e.g. blow away config cache during disaster recovery.
+        vlog(clusterlog.info, "Can't load config cache: {}", e);
+        co_return result;
+    }
+
+    for (const auto& i : config) {
+        ss::sstring key = i.first.as<std::string>();
+        auto& value = i.second;
+        if (key == version_key) {
+            result.version = value.as<config_version>();
+            continue;
+        }
+
+        // Run locally to get result
+        preload_local(key, value, std::ref(result));
+        result.raw_values[key] = YAML::Dump(value);
+
+        // Broadcast value to all shards
+        co_await ss::smp::invoke_on_all(
+          [&key, &value]() { preload_local(key, value, std::nullopt); });
```

Comment:
Because we're doing the actual load one time (before broadcasting the loaded values to all shards), having the values match up should be a given -- if trying to check that, I'd basically be writing the same loop over config settings and invoke_on_all loop but with a check as the innermost thing.